src/Kafka/ConsumerGroups/components/ConsumerGroupTable.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  ActionsColumn,
+  IAction,
+  ISortBy,
+  TableComposable,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  ThProps,
+  Tr,
+} from "@patternfly/react-table";
+import { FunctionComponent, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { ConsumerGroup } from "../types";
+import { ConsumerGroupStateLabel } from "./ConsumerGroupState";
+
+export type ConsumerGroupTableProps = {
+  consumerGroup: ConsumerGroup[];
+  consumerGroupByTopic: boolean;
+  onClickDeleteModal: (data: ConsumerGroup) => void;
+  onClickPartitionoffset: (data: ConsumerGroup) => void;
+  onClickResetoffset: (data: ConsumerGroup) => void;
+};
+
+export const ConsumerGroupTable: FunctionComponent<ConsumerGroupTableProps> = ({
+  consumerGroup,
+  consumerGroupByTopic,
+  onClickDeleteModal,
+  onClickPartitionoffset,
+  onClickResetoffset,
+}) => {
+  const { t } = useTranslation(["kafka"]);
+
+  const [sortBy, setSortBy] = useState<ISortBy>({
+    index: undefined,
+    direction: "asc",
+  });
+
+  const getSortableRowValues = (consumer: ConsumerGroup): string[] => {
+    const { consumerGroupId } = consumer;
+    return [consumerGroupId];
+  };
+
+  consumerGroup.sort((a, b) => {
+    const aValue = getSortableRowValues(a)[sortBy.index || 0];
+    const bValue = getSortableRowValues(b)[sortBy.index || 0];
+    if (sortBy.direction === "asc") {
+      return aValue.localeCompare(bValue);
+    }
+    return bValue.localeCompare(aValue);
+  });
+
+  const getSortParams = (columnIndex: number): ThProps["sort"] => ({
+    sortBy: sortBy,
+    onSort: (_event, index, direction) => {
+      setSortBy({ index, direction });
+    },
+    columnIndex,
+  });
+
+  const tableColumns = {
+    group_id: t("consumerGroup.consumer_group_id"),
+    active_members: t("consumerGroup.active_members"),
+    partitions_with_lag: t("consumerGroup.partitions_with_lag"),
+    state: t("consumerGroup.state_header"),
+  };
+
+  const tableRow = (consumer: ConsumerGroup) => {
+    if (consumerGroupByTopic) {
+      return [];
+    }
+    const actionResolver = (consumer: ConsumerGroup): IAction[] => [
+      {
+        title: t("common:delete"),
+        onClick: () => onClickDeleteModal(consumer),
```

Comment:
It could be a good idea to have a jest test that checks for these callbacks, to see they are getting properly called with the right parameters. Do you know how to do that? Ping me in the chat in case you don't!